tuxemon/menu/quantity.py

```diff
@@ -135,8 +136,10 @@ def initialize_items(self) -> Generator[MenuItem[None], None, None]:
             price_tag = T.translate("shop_buy_free")
         else:
             price_tag = str(price)
-            if self.quantity == 0:
+            if self.quantity == 0 and price > wallet:
```

Comment:
I don't get why we are adding an extra condition here.

So it means that now "soldout" is going to be displayed only if the item quantity is 0, and the price is bigger than the wallet ?
By extension, it means "soldout" would not be displayed if the quantity is 0, but the player has enough money in their wallet.

Reply:
@Grimmys I totally agree with you, but I found really annoying dispay soldout for an item "infinite" like the basic tuxeball. This is sort of a micro "patch", because I would like to add a field inside the class (eg a boolean related to infinite), so we can come up with more precise labels (eg if you are broke, etc.)

Reply:
Oh wait does it mean that quantity = 0 is equivalent to quantity = infinite?
So is it really possible that an item is soldout?

Reply:
It's possible that an item is soldout, but at the moment isn't enabled, because if you exit and re-enter inside the store, and here we go again, the item is returned.

regarding the quantity, I added this `and price > wallet` because of this "issue", let's say you want to buy potions and you play with the quantity by increasing and decreasing it. Well without the  `and price > wallet`  if you selected 0 it appears:
![Screenshot_2023-11-15_12-52-21](https://github.com/Tuxemon/Tuxemon/assets/64643719/a7dd225d-cd26-4a17-8177-d3e1e4b29954)
while after  `and price > wallet`  this is what happens:
![Screenshot_2023-11-15_12-53-14](https://github.com/Tuxemon/Tuxemon/assets/64643719/f00edb02-ff8a-4359-b03b-b7b16c54f7d2)

Reply:
Oh ok, it's just that I thought `quantity` meant something like "stock" (how many instances of the given item are still available in the shop), not the "quantity you want to buy".

My bad for not having read the code more carefully.

Still don't get why in your current example it's displaying $20 for the price instead of $0 (not buying anything should be... free, right?).